### PR TITLE
Add Two Phase Updates feature

### PR DESCRIPTION
**Pull request by astronouth7303**
Felt like writing a two-phase update system.

This is so much easier to use with #263 that it's a soft requirement.

This assumes that events signaled are queued and handled after all of the current event handlers are called.

This skipped our usual discussion phase of things, where we debate designs first. Given that, feel free to completely rip this apart.

### CODE REVIEW

**Comment by astronouth7303** [comment]
umm... let me rebase this quick.

**Comment by astronouth7303** [comment]
This is now a lot more reasonable.

**Comment by astronouth7303** [comment]
That was pretty much the idea? Everyone runs their simulations during `Update`, but stages the changes, so everyone sees a consistent view of the application state.

The simplest usage would be Conway's Game of Life, written such that each cell performs its own logic. Each cell would run its simulation during the `Update` phase and stage the changes, which would be applied during the `Commit` phase, after all other cells have run their simulation.

It did cross my mind to extend this to allow arbitrary phases, so you could have, for example, AI, Physics, and Commit phases. But this seemed like a good, solid start.

File: examples/three_body.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+def setup(scene):
+    scene.add(Planet(position=(3, 0), velocity=Vector(0, 1)))
+    scene.add(Planet(position=(-3, 3), velocity=Vector(1, -1)))
+    scene.add(Planet(position=(-3, -3), velocity=Vector(-1, 0)))
+
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.INFO)
+
+    # Not good practice, https://github.com/ppb/pursuedpybear/issues/263
```
[the remark this answers is not in the document]
Comment (astronouth7303) [reply examples/three_body.py]: Actually not, unless we include the entire kwargs block from `__init__.py`.

[the remark this answers is not in the document]
Comment (astronouth7303) [reply examples/three_body.py]: This is improvable with #307

[the remark this answers is not in the document]
Comment (pathunstrom) [reply examples/three_body.py]: #307 is merged, we can actually fix this now.

**Pull request merged**